include/boost/math/interpolators/catmull_rom.hpp

```diff
@@ -111,9 +111,9 @@ catmull_rom<Point, RandomAccessContainer>::catmull_rom(RandomAccessContainer&& p
     m_pnts[num_pnts+2] = m_pnts[1];
 
     auto tmp = m_pnts[num_pnts-1];
-    for (std::ptrdiff_t i = num_pnts-1; i >= 0; --i)
+    for (size_t i = num_pnts; i != 0; --i)
```

Comment:
Can you change the type to `std::size_t` to match `num_pnts` or make it `auto`?

Reply:
Done